physics/SFC_Models/Land/Noahmp/lnd_iau_mod.F90

```diff
@@ -0,0 +1,915 @@
+   integer, intent(in)                        :: isc, jsc, nx, ny, tile_num, nblks, lsoil, lsnow_lsm
+   integer, dimension(:),          intent(in) :: blksz   !(one:) !GFS_Control%blksz
+   real(kind=kind_phys), intent(in)           :: dtp             !< physics timestep in seconds
+   real(kind=kind_phys), intent(in)           :: fhour           !< current forecast hour
+   character(len=*),              intent(out) :: errmsg
+   integer,                       intent(out) :: errflg
+   
+   integer                                    :: nb, ix
+   integer                                    :: nlunit = 360          ! unit for namelist  !, intent(in)
+   integer                                    :: ios
+   logical                                    :: exists
+   character(len=512)                         :: ioerrmsg
+
+   character(len=:), pointer, dimension(:)    :: input_nml_file => null()
+   character(len=4)                           :: iosstr
+
+   !> land iau setting read from namelist
+   logical               :: do_land_iau               = .false.
+   real(kind=kind_phys)  :: land_iau_delthrs              = 0           !< iau time interval (to scale increments)
+   character(len=240)    :: land_iau_inc_files(7)         = ''          !< list of increment files
+   real(kind=kind_phys)  :: land_iau_fhrs(7)               = -1          !< forecast hours associated with increment files
+   logical               :: land_iau_filter_increments    = .false.     !< filter IAU increments
+
+   integer               :: lsoil_incr = 4
+   logical               :: land_iau_upd_stc = .false.
+   logical               :: land_iau_upd_slc = .false.
+   logical               :: land_iau_do_stcsmc_adjustment = .false.
+   real(kind=kind_phys)  :: land_iau_min_T_increment = 0.0001
+  
+   NAMELIST /land_iau_nml/ do_land_iau, land_iau_delthrs, land_iau_inc_files, land_iau_fhrs,   &  !land_iau_gaussian_inc_file,   &
+                        land_iau_filter_increments, &  
+                        lsoil_incr, land_iau_upd_stc, land_iau_upd_slc, land_iau_do_stcsmc_adjustment, land_iau_min_T_increment                                    
+   
+   !Errors messages handled through CCPP error handling variables
+   errmsg = ''
+   errflg = 0
+
+!3.11.24: copied from GFS_typedefs.F90 
+#ifdef INTERNAL_FILE_NML
+    ! allocate required to work around GNU compiler bug 100886
+    ! https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100886
+    allocate(input_nml_file, mold=input_nml_file_i)
+    input_nml_file => input_nml_file_i
+    read(input_nml_file, nml=land_iau_nml, ERR=888, END=999, iostat=ios)
+#else
+   ! if (file_exist(fn_nml)) then 
+   inquire (file=trim(fn_nml), exist=exists)    ! TBCL: this maybe be replaced by nlunit passed from ccpp
+   if (.not. exists) then
+      write(6,*) 'lnd_iau_mod_set_control: namelist file ',trim(fn_nml),' does not exist'      

```

Comment:
could you not define `errmsg` first and then `write(6,*) errmsg` since they are duplicated?